### PR TITLE
fix the device inconsistency for import convert_sync_batchnorm

### PR DESCRIPTION
**Pull request by jjsjann123**
This fixes the device inconsistency reported in #37930

### CODE REVIEW

**Comment by jjsjann123** [comment]
That's a lot of failure with 2 lines of code change :facepalm: 

From what I can see, the related failure is this one:
```
May 19 19:22:27   test_run_mypy (__main__.TestTypeHints) ... torch/nn/intrinsic/modules/fused.py:45: error: Name 'BatchNorm1d' is not defined  [name-defined]

May 19 19:22:27 torch/nn/intrinsic/modules/fused.py:63: error: Name 'BatchNorm1d' is not defined  [name-defined]

May 19 19:22:42 Found 2 errors in 1 file (checked 980 source files)
```

I don't know how `BatchNorm1d` pops up in the error. I have `BatchNorm2d` in my local script... I can try to import `BatchNorm1d` in that file, but that seems to cover up a stranger problem...
Any thoughts?

File: torch/nn/modules/batchnorm.py [hunk 1]
```diff
@@ -500,8 +500,8 @@ def convert_sync_batchnorm(cls, module, process_group=None):
                                                    process_group)
             if module.affine:
                 with torch.no_grad():
-                    module_output.weight.copy_(module.weight)
-                    module_output.bias.copy_(module.bias)
+                    module_output.weight = module.weight
```
Comment (ngimel) [inline torch/nn/modules/batchnorm.py:503]: out of curiosity, instead of copying member by member, can you do `load_state_dict` thing here that you do in the tests?

File: torch/nn/modules/batchnorm.py [hunk 2]
```diff
@@ -500,8 +500,8 @@ def convert_sync_batchnorm(cls, module, process_group=None):
...
+                    module_output.bias = module.bias
                 # keep requires_grad unchanged
                 module_output.weight.requires_grad = module.weight.requires_grad
```
[the remark this answers is not in the document]
Comment (jjsjann123) [reply torch/nn/modules/batchnorm.py:506]: Yes. IIUC, It's the same Tensor (same storage and same flags). So any change to the old module will be reflected on the new module.

File: torch/nn/modules/batchnorm.py (hunk 1 above)
Comment (jjsjann123) [reply torch/nn/modules/batchnorm.py:503]: Nope. `load_state_dict` is basically a `copy_`. So it would not get the device property appropriately.
It worked in the test case because we explicitly moved the module to cuda to match the original.

File: torch/nn/modules/batchnorm.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ngimel) [reply torch/nn/modules/batchnorm.py:506]: So can you remove these lines then?

**Review by ngimel — APPROVED** [review]
